src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,42 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newFixedThreadPool(5);
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            try {
+                executorService.execute(() -> listener.onEvent(event));
+            } catch (Exception ex) {
+                throw new RuntimeException("Can't notifying listener " + listener, ex);
+            }
+        }
     }
 
     public void shutdown() {
-
+        try {
+            executorService.shutdown();
+            if (!executorService.awaitTermination(10, TimeUnit.SECONDS)) {
+                executorService.shutdownNow();
```

Comment:
I think in most cases we'd want to give the executor as much time as it needs to do all the tasks in the queue and shutdown gracefully, so that we do not have to deal with tasks that did not have time to finish or even start. But here I guess it is up to you